src/PreparedGeometry.cpp

```diff
@@ -28,8 +28,8 @@ PreparedGeometry::PreparedGeometry() :
 {
 }
 
-PreparedGeometry::PreparedGeometry( std::auto_ptr<Geometry> geometry, srid_t srid ) :
-    _geometry( geometry ),
+PreparedGeometry::PreparedGeometry( std::unique_ptr<Geometry>&& geometry, srid_t srid ) :
+    _geometry( geometry.release() ),
```

Comment:
When a function claim ownership of a pointer, the `unique_ptr` is best taken by value:
```c++
PreparedGeometry::PreparedGeometry( std::unique_ptr<Geometry> geometry, srid_t srid ) 
    _geometry( std::move(geometry) ),
```

For a long explanation, you can read https://stackoverflow.com/a/8114913/1728537.

The semantic of `void f(std::unique_ptr<Geometry>&& geometry)` is that `f` *may* take the ownership.

Reply:
I agree, passing rvalue references of unique_ptr seems very confusing to me.